src/main/java/org/jitsi/jicofo/Main.java

```diff
@@ -199,34 +199,29 @@ private static void setSystemProperties(String[] args)
             // For backward compat, the "--domain" command line argument controls the domain for the XMPP component
             // as well as XMPP client connection.
             System.setProperty(XmppClientConnectionConfig.legacyXmppDomainPropertyName, componentDomain);
-            System.setProperty(XmppComponentConfig.domainPropertyName, componentDomain);
         }
         if (host != null)
         {
             // For backward compat, the "--host" command line argument controls the hostname for the XMPP component
             // as well as XMPP client connection.
-            System.setProperty(XmppComponentConfig.hostnamePropertyName, host);
             System.setProperty(XmppClientConnectionConfig.legacyHostnamePropertyName, host);
         }
 
         String componentSubDomain = cmdLine.getOptionValue("--subdomain", "focus");
-        if (componentSubDomain != null)
-        {
-            System.setProperty(XmppComponentConfig.subdomainPropertyName, componentSubDomain);
-        }
-
         int port = cmdLine.getIntOptionValue("--port", 5347);
-        System.setProperty(XmppComponentConfig.portPropertyName, String.valueOf(port));
-
         String secret = cmdLine.getOptionValue("--secret");
         if (isBlank(secret))
         {
             secret = System.getenv("JICOFO_SECRET");
         }
-        if (secret != null)
-        {
-            System.setProperty(XmppComponentConfig.secretPropertyName, secret);
-        }
+
+        XmppComponentConfig.config = new XmppComponentConfig(
+                host == null ? "" : host,
```

Comment:
I don't think I'm following this...how does `host` get pulled from config?  I see we read it from the command line here and pass that if it's non-null.

Reply:
Yes. Currently the only way to configure this value is with command line arguments. Since the aim is to soon remove the component code altogether, I chose not to do "new config" properties for it.

Reply:
Ah! Now I see.  Ok.